config/toml.go

```diff
@@ -17,34 +27,158 @@ func EnsureRoot(rootDir string) {
 
 	configFilePath := path.Join(rootDir, "config.toml")
 
-	// Write default config file if missing.
-	if !cmn.FileExists(configFilePath) {
-		// Ask user for moniker
-		// moniker := cfg.Prompt("Type hostname: ", "anonymous")
-		cmn.MustWriteFile(configFilePath, []byte(defaultConfig("anonymous")), 0644)
+	if cmn.FileExists(configFilePath) {
+		// TODO: Prompt user to replace if config file already exists.
+	} else {
+		// Ask user for moniker [moniker := cfg.Prompt("Type hostname: ", "anonymous")] (TODO)
+		// Then write the default config from template
+		writeConfigFile(configFilePath)
+	}
+}
+
+// XXX: this func should probably be called by cmd/tendermint/commands/init.go
+// alongside the writing of the genesis.json and priv_validator.json
+func writeConfigFile(configFilePath string) {
+	var buffer bytes.Buffer
+
+	if err := configTemplate.Execute(&buffer, DefaultConfig()); err != nil {
+		panic(err)
 	}
+
+	cmn.MustWriteFile(configFilePath, buffer.Bytes(), 0644)
 }
 
-var defaultConfigTmpl = `# This is a TOML config file.
+const defaultConfigTemplate = `# This is a TOML config file.
 # For more information, see https://github.com/toml-lang/toml
 
-proxy_app = "tcp://127.0.0.1:46658"
-moniker = "__MONIKER__"
-fast_sync = true
-db_backend = "leveldb"
-log_level = "state:info,*:error"
+##### main base config options #####
```

Comment:
ok, this is amazing, but here's my crazy question .... can we get this code autogenerated from the config struct during a build process so we don't have to worry about redundant commentary here and in the structs??  is that a crazy thing to ask for :P

Reply:
maybe, in the future? I agree with the sentiment but have noticed that generally autogenerated codestuffs get mangled in one way or another. I'll add notices in the appropriate locations.